gnucash/register/register-gnome/completioncell-gnome.c

```diff
@@ -0,0 +1,910 @@
+    PopBox* box = cell->cell.gui_private;
+
+    gnc_completion_cell_gui_destroy (& (cell->cell));
+
+    if (box)
+    {
+        if (box->item_hash)
+            g_hash_table_destroy (box->item_hash);
+
+        g_free (box);
+        cell->cell.gui_private = NULL;
+    }
+    cell->cell.gui_private = NULL;
+    cell->cell.gui_realize = NULL;
+}
+
+//FIXME could just use builtin, maybe needed for direction?
+static gint
+my_sort_func (GtkTreeModel* model, GtkTreeIter* iter_a, GtkTreeIter* iter_b, gpointer user_data)
+{
+    gint a_weight, b_weight;
+    gint ret = 0;
+
+    gtk_tree_model_get (model, iter_a, WEIGHT_COL, &a_weight, -1);
+    gtk_tree_model_get (model, iter_b, WEIGHT_COL, &b_weight, -1);
+
+    if (a_weight < b_weight)
+        ret = -1;
+    else if (b_weight < a_weight)
+        ret = 1;
+
+    return ret;
+}
+
+void
+gnc_completion_cell_set_sort_enabled (CompletionCell* cell,
+                                      gboolean enabled)
+{
+    if (!cell)
+        return;
+
+    PopBox* box = cell->cell.gui_private;
+    box->sort_enabled = enabled;
+}
+
+
+static void
+set_sort_column_enabled (PopBox* box, gboolean enable)
+{
+    if (enable)
+    {
+        gtk_tree_sortable_set_sort_func (GTK_TREE_SORTABLE(box->item_list->list_store),
+                                         WEIGHT_COL, my_sort_func, box->item_list, NULL);
+
+        gnc_item_list_set_sort_column (box->item_list, WEIGHT_COL);
+    }
+    else
+        gnc_item_list_set_sort_column (box->item_list, GTK_TREE_SORTABLE_UNSORTED_SORT_COLUMN_ID);
+
+}
+
+static void
+item_store_clear (CompletionCell* cell)
+{
+    if (!cell)
+        return;
+
+    PopBox* box = cell->cell.gui_private;
+
+    // disconnect list store from tree view
+    GtkListStore *store = gnc_item_list_disconnect_store (box->item_list);
+
+    block_list_signals (cell);
+
+    if (box->sort_enabled) // if sorting, disable it
+        set_sort_column_enabled (box, FALSE);
+
+    box->stop_searching = FALSE;
+    gtk_list_store_clear (box->item_store);
+
+    if (box->sort_enabled) // if sorting, enable it
+        set_sort_column_enabled (box, TRUE);
+
+    unblock_list_signals (cell);
+
+    // reconect list store to tree view
+    gnc_item_list_connect_store (box->item_list, store);
+
+    gnc_item_edit_hide_popup (box->item_edit);
+    box->list_popped = FALSE;
+}
+
+void
+gnc_completion_cell_clear_menu (CompletionCell* cell)
+{
+    if (!cell)
+        return;
+
+    PopBox* box = cell->cell.gui_private;
+    if (!box)
+        return;
+
+    if (box->item_list)
+    {
+        g_hash_table_remove_all (box->item_hash);
+        item_store_clear (cell);
+    }
+}
+
+void
+gnc_completion_cell_add_menu_item (CompletionCell* cell,
+                                   const char* menustr,
+                                   time64 date)
+{
+    if (!cell)
+        return;
+    if (!menustr)
+        return;
+
+    PopBox* box = cell->cell.gui_private;
+
+    if (box->item_list)
```

Comment:
Should this be `box->item_hash` ? You're not using `box->item_list` in the code that follows, so I'm unsure why you would skip adding descriptions to the hash if the list item is not set ?

Reply:
I think this was a copy paste error, changed.